lib/codegen/model.ejs

```diff
@@ -24,20 +24,29 @@ function printMethod(modelName) {
     return method + params.join(', ') + ') {';
 }
 
+function joinParams(params) {
+  const newParams = params.map(param => `normalize(${param})`);
+  return newParams.join(',');
+}
+
 function printOperation() {
     var serviceName = '_soapModel';
     var params = op.accepts.map(function(a) {
         return a.arg;
     });
     var method = serviceName + '.' + op.operationId
-            + '(' + params.join(', ') + ', function (err, response) {';
+            + '(' + joinParams(params) + ', function (err, response) {';
     return method;
 }
 
 %>
 'use strict';
 var server = require('../../server/server');
 
+function normalize(parameter) {
+  return parameter.toJSON ? parameter.toJSON() : parameter;
```

Comment:
Maybe we should handle the case where `parameter` is an array.

`typeof parameter.toJSON === 'function'` is probably even better.

Reply:
I will create another web service where I pass an array as input parameter, and receive an array of some object. I will see if it changes the generated output, and update my PR accordingly.

Reply:
@raymondfeng , I actually looked at `normalize(parameter)` called from 

```
_soapModel.getSomeData(normalize(getSomeData), function (err, response) {
```
and `getSomeData` from `normalize(getSomeData)` is never an array. It is a model object
that wraps the actual input parameter of the operation being called; regardless of whether my input parameter is:

myParam : string 

or 

myParam: [string]

Reply:
I can add a comment to the `normalize` function that it is only meant to be called from 

```
_soapModel.{operation}(params, function(err, response){
```
that are generated.